riscv_isa.cpp

```diff
@@ -949,3 +955,117 @@ void ac_behavior(FSW){
   DM.write(RB[rs1] + sign_ext, RBF[rs2]);
   dbg_printf("addr: %d\n\n", RB[rs1]+sign_ext);		
 }
+
+
+void ac_behavior(FADD_S){
+  dbg_printf("FADD.S r%d, r%d, r%d\n", rd, rs1, rs2);
+  dbg_printf("RBF[rs1] = %.3f\n", (float)RBF[rs1]);
+  dbg_printf("RBF[rs2] = %.3f\n", (float)RBF[rs2]);
+  RBF[rd] = RBF[rs1] + RBF[rs2];
```

Comment:
This won't work as intended because RBF is an integer array, so this is essentially adding two integer numbers instead of two floats. You need to first cast it to the appropriate type, perform the operation and then cast it back to integer to store it in the register bank. See https://github.com/rafaelauler/mips/blob/mips32r2/mips_isa.cpp#L1283 for reference on doubles or https://github.com/rafaelauler/mips/blob/mips32r2/mips_isa.cpp#L215 for reference on floats.

To understand the code I linked, see the definition for load_float/load_double/save_float/save_double helper functions here: https://github.com/rafaelauler/mips/blob/mips32r2/mips_isa_helper.H